### PR TITLE
fix(#9315): api v3 post, put, del JSON

### PR DESCRIPTION
**Pull request by pitaj**
also allow `app.alertError` to be called without an argument
also fix `./nodebb build --dev` to actually build in dev mode

### CODE REVIEW

File: public/src/modules/api.js [hunk 1]
```diff
@@ -55,7 +55,8 @@ define('api', () => {
 	api.post = (route, payload, onSuccess) => call({
 		url: route,
 		method: 'post',
-		data: payload,
+		data: JSON.stringify(payload),
+		contentType: 'application/json; charset=utf-8',
```
Comment (julianlam) [inline public/src/modules/api.js:59]: Was it not possible to put this inside `call` if data is passed in?

**Review by julianlam — COMMENTED** [review]
Thanks for the pr 🙂

File: public/src/modules/api.js (hunk 1 above)
Comment (pitaj) [reply public/src/modules/api.js:59]: This seemed like the best way. Don't have to make conditional logic, don't have to worry about messing up get/head, it's more declarative, and it follows the repetition of the csrf.

**Review by julianlam — APPROVED** [review]

**Comment by julianlam** [comment]
Thank you for this :smile:

**Pull request merged**